ec_schnorr.go

```diff
@@ -41,13 +41,24 @@ func (ecs *ECSchnorr) GetPublicKey(privKey []byte, compress bool) []byte {
 }
 
 // Sign returns the signature (r, s) on a given message.
+func (ecs *ECSchnorr) Sign(privKey, pubKey, msg []byte) ([]byte, []byte) {
+	for {
+		k, _ := generateDRN(msg)
```

Comment:
I think this error should be returned.

Reply:
I don't think so... If trySign returns error, new k will be generated and do trySign again, so error never occurs for Sign. I implemented so that Sign function always returns signature successfully.

Reply:
In this function, If it returns error, k will be nil. 
Is it OK to input nil as k of trySign?

```
func (h *HmacDRBG) Generate(byteLength int32, input []byte) ([]byte, error) {
	if h.reseedCounter > reseedInterval {
		return nil, errors.New("Reseed is reqired")
	}

	if len(input) > 0 {
		h.update(input)
	}

	temp := make([]byte, 0, byteLength)
	for i := int32(0); i < byteLength; i += int32(32) {
		mac := hmac.New(sha256.New, h.k)
		mac.Write(h.v)
		h.v = mac.Sum(nil)
		temp = append(temp, h.v...)
	}
	result := make([]byte, byteLength)
	copy(result, temp[:byteLength])

	h.update(input)
	h.reseedCounter++

	return result, nil
}
```

Reply:
I think it depends on the behavior when nil will set to `SetBytes()`.

Reply:
Oh you mean this part "k, _"? Sorry I misunderstood. I agree.